internal/ctxval/acl.go

```diff
@@ -0,0 +1,31 @@
+package ctxval
+
+import (
+	"context"
+)
+
+// RbacAcl is a copy of clients.RbacAcl (import cycle)
+type RbacAcl interface {
+	IsAllowed(res, verb string) bool
+}
+
+// a copy of clients.NoPermissionsAcl (import cycle)
+type noPermAcl struct{}
+
+func (r noPermAcl) IsAllowed(res, verb string) bool {
+	return false
+}
+
+// RbacAcl returns ACL interface, when no ACL is present it returns a list that always evaluates to false.
+func Acl(ctx context.Context) RbacAcl {
+	value := ctx.Value(aclCtxKey)
+	if value == nil {
+		return noPermAcl{}
+	}
+	return value.(RbacAcl)
+}
+
+// WithAcl returns context copy with ACL interface.
+func WithAcl(ctx context.Context, id RbacAcl) context.Context {
+	return context.WithValue(ctx, aclCtxKey, id)
+}
```

Comment:
This ctxval package is used literally EVERYWHERE so basically it cannot import pretty much anything because of import cycles.

This is the reason why I make copies of both the ACL interface and NoPermissionsAcl variable here. This is fine to do in Go, it is better to copy sometimes than to do hacks.

Reply:
Makes me thinking: is this actually the best approach?
Shouldn't we instead make the `WithXXX` functions part of the package they store the value of?
It sounds like it would fix the issue and to me it sounds a bit cleaner.
Here we could define the context key type and so on, but storing to context would be done in each package.
We need to do it for stubs already anyway.

Reply:
I mean I'm fine here, we can decide on and do this later :+1:

Reply:
Yeah that is a valid option I considered, main reason why I like centralized approach is that you don’t need to think about the package name, you just type `ctxval` and the required function will pop up. We use it everywhere in the code.

The split will be **really painful tho** :-) Not here, not now.